### ISSUE TITLE
Ag-Grid Compatability

### ISSUE THREAD

**Issue by MI6-613** [mi6/ic-ui-kit#2130]
When using a custom cell renderer in ag-grid ICDS components do not resize correctly when resizing columns
If you use the tutorial in this [link](https://www.ag-grid.com/react-data-grid/component-cell-renderer/) you can see they have implemented cell renderers that when resized work. 
If you were to change the website url cell so that it renderer inside a MUI chip you would find the chip resizes to fit the width of the cell and does ... etc if the cell size is smaller than the content. 
However if you use ICDS Chip or typography the behaviour when resizing is akin to cropping the chip or text to the size of the cell rather than resizing the component.

As ag-grid is a very popular table component it would be great if ICDS components could work better in custom cell renderers. I expect the issue is around the shadow dom and the component or ag-grid unaware of the size of the component/cell but I am unsure.

**Comment by MI6-255** [comment]
Initially focus on Chip, if there's more components this will affect, open an another issue for this

**Comment by GCHQ-Developer-299** [comment]
Suggestion: to test fixes for this we could install ag-grid into our React as a dev dependency, and set up a Storybook inside IcChip and IcTypography

**Comment by GCHQ-Developer-847** [comment]
Just so it's clear, the behaviour when using a MUI chip is to truncate the text with an ellipsis, e.g.:

https://github.com/user-attachments/assets/83b48f27-97ff-40ab-87ae-074800ad3902

**Comment by MI6-613** [comment]
> Initially focus on Chip, if there's more components this will affect, open an another issue for this

I believe the issue is with the IcTypography so if you can fix it there it should fix for a large range of components

**Comment by GCHQ-Developer-847** [comment]
Just commenting to explain my solution for those reviewing my PR

Exploration:
- I first tried making the chips / typography truncate with an ellipsis all of the time - when it can't fit within its container. However, this was overriding the current wrapping behaviour, e.g. shown in the [long label chip story](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-chip--long-label).
- To resolve this ^, I tried creating a `maxLines` prop on the chip which would behave like the prop on the tooltip component. Developers would have then been able to apply this and set it to one line, or however many lines of the chip they would want to be visible. This worked fine when outside of an AG Grid but I could not get it working when inside an AG Grid at all (I think maybe due to the AG Grid having lots of inbuilt styling which was overriding things, and I couldn't work out how to solve it).
- Also, with the typography component, using the existing `maxLines` prop wouldn't really work because it would show the 'See more' button and the rows in an AG Grid are a set height so it would be difficult to make displaying the whole piece of text work.

Basically, there were a lot of complications, which led me to believe that the best solution was to make the truncation on the chip and typography components happen only when in an AG Grid. This means there are no conflicts with behaviour with these component which exists when they are not in an AG Grid.

I haven't added tooltips when the chip label or typography text is truncated as I wasn't really sure about whether to add tooltips; I was thinking it could be up to the developer to add tooltips themselves if they want them, and the user always has the option to increase the column size anyway to see the whole value that is in a cell. Happy if hear any other opinions!

**Comment by GCHQ-Developer-847** [comment]
@MI6-613 Both typography and chips should now be truncated when used in an AG Grid (after the next release). 

Unfortunately I wasn't able to find a change to the typography component which would make the truncation work in every component that uses it, so making all components compatible with AG Grids will involve making changes to each individual component (if it needs changing). I will be testing different components in AG Grids and making tickets for them if they have this same issue.
